### PR TITLE
Update test project dependencies

### PR DESCRIPTION
**Pull request by bording**
As part of preparing for #1318, this updates the unit test dependencies to the latest versions, which support .NET Standard targets.

### CODE REVIEW

**Comment by bording** [comment]
Marking as [WIP] until I get the CI builds sorted out. Need to get them to use a newer version of NuGet.

File: LibGit2Sharp/LibGit2Sharp.csproj [hunk 1]
```diff
@@ -11,7 +11,7 @@
     <AppDesignerFolder>Properties</AppDesignerFolder>
     <RootNamespace>LibGit2Sharp</RootNamespace>
     <AssemblyName>LibGit2Sharp</AssemblyName>
-    <TargetFrameworkVersion>v4.0</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.6</TargetFrameworkVersion>
```
Comment (AArnott) [inline LibGit2Sharp/LibGit2Sharp.csproj]: We should be able to revert this. We can target net40 in the library and target net46 in the test project.

Comment (bording) [inline LibGit2Sharp/LibGit2Sharp.csproj]: You're right, that should be possible. Once I finally get the tests working on Travis, I'll go back and change that back.

**Comment by bording** [comment]
And we're green!

I had to disable xUnit 2.x's test parallelization because it was causing file locking conflicts. It would be worth revisiting this in the future to speed up the test runs.

I also had to change up some tests that were using the `InlineData` attribute to pass in `IEnumerable<string>` values. On mono, it was failing with a error about not being able to convert from `System.Object[]` to `IEnumerable<string>`.

**Comment by AArnott** [comment]
Thanks!

**Pull request merged**